src/components/document/document-workspace.tsx

```diff
@@ -35,22 +43,90 @@ export class DocumentWorkspaceComponent extends BaseComponent<IProps> {
       removeLoadingMessage("Building workspace");
+      const loadingMeasurements = getLoadingMeasurements();
+      this.primaryDocument = this.getPrimaryDocument(this.stores.persistentUI.problemWorkspace.primaryDocumentKey);
+      this.setState({ loadingMeasurements });
+      this.logLoadingAndDocumentMeasurements();
+
+    });
+  }
+
+  public logLoadingAndDocumentMeasurements(){
+    const startTime = performance.now();
+    const { documents, problem: { sections }, teacherGuide } = this.stores;
+    const { loadingMeasurements } = this.state.loadingMeasurements;
+
+    const totalNumDocumentsLoaded = documents.all.length;
+    const totalNumTilesLoaded = documents.all.reduce((total: number, doc: DocumentModelType) => {
+      return total + (doc.content?.tileMap.size || 0);
+    }, 0);
+
+    const primaryDocTilesByType = this.primaryDocument?.content?.getAllTilesByType() as any;
+    const primaryDocNumTilesByType = countTileKeys(primaryDocTilesByType);
+
+    //getAllTilesByType returns a map with the tileKeys as values, we want to convert this to the length
+    function countTileKeys(tilesByType: Record<string, string[]>): Record<string, number> {
+      const tileCounts: Record<string, number>= {};
+      for (const tileType of Object.keys(tilesByType)) {
+        tileCounts[tileType] = tilesByType[tileType].length;
+      }
+      return tileCounts;
+    }
+
+    // ----------------------- Curriculum Documents Summary  --------------------------------------
+    // Take into account that teachers have extra "curriculum documents" in the TeacherGuide tab
+    let curriculumDocSections = [...sections]; //these are for the "Problem" tab
+    if(teacherGuide) {
+      curriculumDocSections = [...curriculumDocSections, ...teacherGuide.sections];
+    }
+
+    const curriculumSectionsTilesByType = curriculumDocSections.map((section) => {
+      const sectionDocTilesByType = section.content?.getAllTilesByType() as any;
+      const sectionDocNumTilesByType = countTileKeys(sectionDocTilesByType);
+      return sectionDocNumTilesByType;
+    });
+
+    //Convert curriculumSectionsTileByType further and count all tiles in each section
+    const curriculumSumTileTypes: Record<string, number> = {};
+    curriculumSectionsTilesByType.forEach((section) => {
+      Object.keys(section).forEach((tileType) => {
+        if(!curriculumSumTileTypes[tileType]) {
+          curriculumSumTileTypes[tileType] = 0; //Create entry
+        }
+        curriculumSumTileTypes[tileType] += section[tileType];
+      });
     });
+
+    const documentMeasurements = {
+      totalNumDocumentsLoaded,
+      totalNumTilesLoaded,
+      primaryDocNumTilesByType,
+      curriculumSumTileTypes,
+    };
+
+    const finalLogObject = {
+      loadingMeasurements,
+      documentMeasurements
+    };
+
+    const endTime = performance.now();
+    console.log(`logLoadingAndDocumentMeasurements executed in ${endTime - startTime} milliseconds`);
```

Comment:
Were you thinking we should add the calculation time ( `endTime - startTime`) as a property to the `finalLogObject`? Or just for us to analyze it's impact on the loading of CLUE?

Reply:
I think it is fine to just put it in the console when the "loading" debug flag is on. If we find through testing that it takes a lot of time, then we are going to have to fix it or take it out.